samza-core/src/test/java/org/apache/samza/zk/TestZkBarrierForVersionUpgrade.java

```diff
@@ -74,8 +73,7 @@ public static void teardown() {
     zkServer.teardown();
```

Comment:
zkServer.setup is called in Before, but zkServer.teardown is called in AfterClass. You may want to call zkServer.teardown in After for each test, like  zkServer.setup in Before, or make zkServer.setup called in BeforeClass.